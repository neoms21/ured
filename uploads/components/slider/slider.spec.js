import React from "react";
import ReactTestRenderer from "react-test-renderer";
import Slider from "./slider";

import { mount } from "enzyme";

describe("Slider tests", () => {
  it("should render the slider component", () => {
    const items = [
      { id: 1, label: "11" },
      { id: 2, label: "11" },
      { id: 3, label: "11" }
    ];
    const component = ReactTestRenderer.create(<Slider items={items} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should set the state with respect to id of div clicked", () => {
    const changeMock = jest.fn();
    const props = {
      items: [
        { id: 1, label: "11" },
        { id: 2, label: "11" },
        { id: 3, label: "11" }
      ],
      input: { onChange: changeMock }
    };
    const component = mount(<Slider {...props} />);
    const item = component.find("#step-2");

    item.simulate("click", { currentTarget: { offsetLeft: 100 } });

    expect(component.state()).toEqual({
      left: -7,
      position: "absolute",
      selected: true,
      selectedItem: "2"
    });

    expect(changeMock).toHaveBeenCalledWith("2");
  });

  it("should set the state with respect to id of div clicked and not call input on change if not form element", () => {
    const changeMock = jest.fn();
    const props = {
      items: [
        { id: 1, label: "11" },
        { id: 2, label: "11" },
        { id: 3, label: "11" }
      ]
    };
    const component = mount(<Slider {...props} />);
    const item = component.find("#step-2");

    item.simulate("click", { currentTarget: { offsetLeft: 100 } });

    expect(component.state()).toEqual({
      left: -7,
      position: "absolute",
      selected: true,
      selectedItem: "2"
    });

    expect(changeMock).not.toHaveBeenCalled();
  });
});
