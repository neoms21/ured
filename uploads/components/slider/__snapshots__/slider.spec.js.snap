// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider tests should render the slider component 1`] = `
<div
  className="container"
>
  <h3>
     
  </h3>
  <img
    alt="marker"
    onClick={undefined}
    src="images/slider-button.svg"
    style={
      Object {
        "left": undefined,
        "position": "absolute",
        "top": "90px",
        "transition": "left .1s linear",
        "zIndex": 99999,
      }
    }
  />
  <div
    className="slider"
  >
    <a
      className="step"
      id="step-1"
      onClick={[Function]}
    >
      <div
        id="left"
      />
      <div
        className="divider"
      />
      <span
        className="stepLabel"
      >
        11
      </span>
      <div
        id="right"
      />
    </a>
    <a
      className="step"
      id="step-2"
      onClick={[Function]}
    >
      <div
        id="left"
      />
      <div
        className="divider"
      />
      <span
        className="stepLabel"
      >
        11
      </span>
      <div
        id="right"
      />
    </a>
    <a
      className="step"
      id="step-3"
      onClick={[Function]}
    >
      <div
        id="left"
      />
      <div
        className="divider"
      />
      <span
        className="stepLabel"
      >
        11
      </span>
      <div
        id="right"
      />
    </a>
  </div>
</div>
`;
